tests/pytests/unit/utils/test_network.py

```diff
@@ -1,8 +1,1547 @@
+            "ret": "127.0.0.1",
+        },
+    ]
+    for host in hosts:
+        with patch.object(
+            socket,
+            "getaddrinfo",
+            create_autospec(socket.getaddrinfo, return_value=host["mocked"]),
+        ):
+            with patch("socket.socket", create_autospec(socket.socket)):
+                ret = network.dns_check(host["host"], host["port"])
+                assert ret == host["ret"]
+
+
+def test_dns_check_ipv6_filter():
+    # raise exception to skip everything after the getaddrinfo call
+    with patch.object(
+        socket,
+        "getaddrinfo",
+        create_autospec(socket.getaddrinfo, side_effect=Exception),
+    ) as getaddrinfo:
+        for ipv6, param in [
+            (None, socket.AF_UNSPEC),
+            (True, socket.AF_INET6),
+            (False, socket.AF_INET),
+        ]:
+            with pytest.raises(Exception):
+                network.dns_check("foo", "1", ipv6=ipv6)
+            getaddrinfo.assert_called_with("foo", "1", param, socket.SOCK_STREAM)
+
+
+def test_dns_check_errors():
+    with patch.object(
+        socket, "getaddrinfo", create_autospec(socket.getaddrinfo, return_value=[])
+    ):
+        with pytest.raises(
+            salt.exceptions.SaltSystemExit,
+            match="DNS lookup or connection check of 'foo' failed.",
+        ) as exc_info:
+            network.dns_check("foo", "1")
+
+    with patch.object(
+        socket,
+        "getaddrinfo",
+        create_autospec(socket.getaddrinfo, side_effect=TypeError),
+    ):
+        with pytest.raises(
+            salt.exceptions.SaltSystemExit, match="Invalid or unresolveable address"
+        ) as exc_info2:
+            network.dns_check("foo", "1")
+
+
+def test_test_addrs():
+    # subset of real data from getaddrinfo against saltstack.com
+    addrinfo = [
+        (30, 2, 17, "", ("2600:9000:21eb:a800:8:1031:abc0:93a1", 0, 0, 0)),
+        (30, 1, 6, "", ("2600:9000:21eb:a800:8:1031:abc0:93a1", 0, 0, 0)),
+        (30, 2, 17, "", ("2600:9000:21eb:b400:8:1031:abc0:93a1", 0, 0, 0)),
+        (30, 1, 6, "", ("2600:9000:21eb:b400:8:1031:abc0:93a1", 0, 0, 0)),
+        (2, 1, 6, "", ("13.35.99.52", 0)),
+        (2, 2, 17, "", ("13.35.99.85", 0)),
+        (2, 1, 6, "", ("13.35.99.85", 0)),
+        (2, 2, 17, "", ("13.35.99.122", 0)),
+    ]
+    with patch("socket.socket", create_autospec(socket.socket)) as s:
+        # we connect to the first address
+        addrs = network._test_addrs(addrinfo, 80)
+        assert len(addrs) == 1
+        assert addrs[0] == addrinfo[0][4][0]
+
+        # the first lookup fails, succeeds on next check
+        s.side_effect = [socket.error, MagicMock()]
+        addrs = network._test_addrs(addrinfo, 80)
+        assert len(addrs) == 1
+        assert addrs[0] == addrinfo[2][4][0]
+
+        # attempt to connect to resolved address with default timeout
+        s.side_effect = socket.error
+        addrs = network._test_addrs(addrinfo, 80)
+        time.sleep(2)
```

Comment:
```suggestion
        time.sleep(2)
```
Do we really need to slow down the test?

Reply:
I only ported it, not write it :laughing: , will take a look

Reply:
done